Code/Framework/AzCore/AzCore/Metrics/IEventLogger.h

```diff
@@ -8,127 +8,329 @@
 
 #pragma once
 
-#include <type_traits>
 #include <AzCore/base.h>
-#include <AzCore/RTTI/TypeInfoSimple.h>
-#include <AzCore/std/limits.h>
+#include <AzCore/Outcome/Outcome.h>
+#include <AzCore/Platform.h>
+#include <AzCore/RTTI/RTTI.h>
+#include <AzCore/std/chrono/chrono.h>
+#include <AzCore/std/containers/span.h>
+#include <AzCore/std/containers/variant.h>
 #include <AzCore/std/string/string_view.h>
-#include <AzCore/Casting/numeric_cast.h>
+#include <AzCore/std/utils.h>
+
+namespace AZ::IO
+{
+    class GenericStream;
+}
 
 namespace AZ::Metrics
 {
-    //! Simple hash structure based on DJB2a to generate event IDs at compile time
-    class EventNameHash
+    constexpr AZStd::string_view ArgsKey = "args";
+
+    //! Event fields structure that can be used to record event argument data
+    //! NOTE: the EventValue does not own any data
+    //! Any string or array data must outlive the EventValue
+    struct EventValue;
+
+    //! Encapsulates an EventField paired with a field name.
+    //! Represents a single json key, value pair field
+    struct EventField;
+
+    //! std::span requires a complete type, so an EventValue pointer and a size member
```

Comment:
do we want to refer to ```std::span``` or ```AZStd::span``` even if its just an alias?

Reply:
Well I always debate which one to mention.
In this case I mean to imply that the `std::span` class which we have a conforming standard implementation in the AZStd namespace is what requires a complete type